diff --git a/lib/esi.js b/lib/esi.js
index c33bd69..cc2f855 100644
--- a/lib/esi.js
+++ b/lib/esi.js
@@ -11,8 +11,7 @@ function ESI(config) {
     const onError = config.onError || (() => {});
     const dataProvider = config.dataProvider || new DataProvider(config);
     const logger = new Logger(config);
-    const allowedHosts = (typeof config.allowedHosts === 'object') ? config.allowedHosts :
-        new AllowedHosts(config, logger);
+    const allowedHosts = new AllowedHosts(config, logger);
 
     function processHtmlText(html, options, state) {
         options = options || {};
