diff -u a/app/config.js b/app/config.js
--- a/app/config.js	2018-10-26 11:10:49.150979950 -0400
+++ b/app/config.js	2018-10-26 15:24:17.251557293 -0400
@@ -52,6 +52,7 @@
     try {
         confObj = require(fullPath);
     } catch (e) {
+        console.log(e);
         throw new Error(
             `Tried to open spandx config file ${c.fg.l.cyan}${filePath}${
                 c.end
diff -u a/app/router.js b/app/router.js
--- a/app/router.js	2018-10-26 11:10:49.150979950 -0400
+++ b/app/router.js	2018-10-31 13:12:58.333473120 -0400
@@ -86,15 +86,26 @@
             );
         }
 
-        if (target) {
-            proxy.web(req, res, { target }, e => {
-                console.error(e);
-                res.writeHead(200, { "Content-Type": "text/plain" });
+        const doProxy = function (t) {
+            target = t || target;
+            if (target) {
+                proxy.web(req, res, { target }, e => {
+                    console.error(e);
+                    res.writeHead(200, { "Content-Type": "text/plain" });
+                    res.end();
+                });
+            } else {
+                res.writeHead(404);
                 res.end();
+            }
+        };
+
+        if (conf.hasOwnProperty('routerPlugin') && typeof conf.routerPlugin === 'function') {
+            conf.routerPlugin(req, res, target).then((target) => {
+                doProxy(target);
             });
         } else {
-            res.writeHead(404);
-            res.end();
+            doProxy();
         }
     };
 };
