diff --git a/app/spandx.js b/app/spandx.js
index 51768c4..be0f993 100644
--- a/app/spandx.js
+++ b/app/spandx.js
@@ -210,7 +210,7 @@ function init(confIn) {
                                 c.fg.l.green
                             }${route[0]}${c.e} will be routed to ${
                                 c.fg.l.blue
-                            }${route[1].host}${c.e}${c.fg.l.green}${route[0]}${
+                            }${route[1].host.default || route[1].host}${c.e}${c.fg.l.green}${route[0]}${
                                 c.e
                             }`
                     )
